libkbfs/test_stallers.go

```diff
@@ -38,6 +39,166 @@ const (
 	StallableMDPutUnmerged           StallableMDOp = "PutUnmerged"
 )
 
+type naïveStallContext struct {
+	onStalled <-chan struct{}
+	unstall   chan<- struct{}
+	undoStall func()
```

Comment:
Not sure if having an arbitrary function is the best way to do this -- if you had:

```
type naiveBlockStall struct {
  onStalled ...
  unstalled ...
  oldBlockOps BlockOps
}
```

then you can move the logic to `UndoStallBlockOp`, which seems clearer. (and similarly for naiveMDStall).